shared/test/test_assets.rb

```diff
@@ -56,7 +56,17 @@ def test_assets
     delete(@assets, channel_id, 'filename.jpg')
     assert @assets.last_response.successful?
 
-    # invalid files are not uploaded
+    # file extension case insensitivity
+    put(@assets, channel_id, 'filename.JPG', 'stub-contents', 'application/jpeg')
+    assert @assets.last_response.successful?
+    get(@assets, channel_id, 'filename.JPG')
+    assert @assets.last_response.successful?
+    get(@assets, channel_id, 'filename.jpg')
+    assert @assets.last_response.not_found?
```

Comment:
Do we want to just save the filename with a lowercase extension?  I can't think of a time where we'd prefer to support different casing of extensions.